nucliadb/nucliadb/search/search/query.py

```diff
@@ -403,13 +407,19 @@ async def parse_vector_search(self, request: nodereader_pb2.SearchRequest) -> bo
                     incomplete = True
         else:
             query_vector = self.user_vector
+
+        # XXX: Do we need some vectorset validation?
```

Comment:
You'll probably need to detect if the vectorset provided exists (or catch the situation where the key is not found) and raise a proper http 4xx

Reply:
passing the vectoset here is missing too:

https://github.com/nuclia/nucliadb/pull/2210/files#diff-277e84d03ac037bc7b1d21d8811c5506692cb40139ab8b34b7ef27151723c083L222

Reply:
Added vectorset validation and raise `InvalidQueryError` if it's not found